src/python/pants/backend/python/interpreter_cache.py

```diff
@@ -115,18 +115,19 @@ def select_interpreter_for_targets(self, targets):
     # Return the lowest compatible interpreter.
     return min(allowed_interpreters)
 
-  def _interpreter_from_path(self, path, filters=()):
-    try:
-      executable = os.readlink(os.path.join(path, 'python'))
-      if not os.path.exists(executable):
-        if os.path.dirname(path) == self._cache_dir:
+  def _interpreter_from_relpath(self, path, filters=()):
+    path = os.path.join(self._cache_dir, path)
+    if os.path.isdir(path):
```

Comment:
You can ditch this check, the `os.readlink` below will already raise OSError if any part of the path DNE.